lib/internal/event_target.js

```diff
@@ -417,6 +423,10 @@ function validateEventListenerOptions(options) {
   };
 }
 
+function isEventTarget(obj) {
+  return obj && obj.constructor && obj.constructor[kIsEventTarget];
```

Comment:
It's not clear how fool-proof you want to be here, but I'll note that people can fool this brand check. For example, `EventTarget.prototype.dispatchEvent.call({ constructor: EventTarget })` would pass with this implementation, but fail per spec.

But anything you do will probably be a compromise. If something works cross-realm, then it can be fooled (and thus is not spec-compliant). If it doesn't work cross-realm, then it is not spec compliant either.

So without C++ there will always be compromises. Maybe this version is a fine compromise.

Reply:
Nit: We should likely include this explanation in a comment here on `isEventTarget()` in case this ends up coming up in a question later.